tests/FSharp.Compiler.ComponentTests/Conformance/Expressions/ObjectExpressions/ObjectExpressions.fs

```diff
@@ -16,21 +16,50 @@ type IFirst =
 
 let x = { new _ with member this.MyMember() = 42 }
         """
-         |> withLangVersion80
+         |> withLangVersion90
          |> typecheck
          |> shouldFail
          |> withDiagnostics [
              (Error 772, Line 5, Col 11, Line 5, Col 16, "'new' must be used with a named type")
          ]
          
+    [<Fact>]
+    let ``Object expression can not implement a class class end`` () =
+        Fsx """
+type Class() = class end
+
+let implementer = { new Class()  }
+        """
+         |> withLangVersion80
```

Comment:
Should be also langVersion90 here I guess?

Reply:
Done